redash/handlers/query_results.py

```diff
@@ -232,6 +233,27 @@ def get(self, query_id=None, query_result_id=None, filetype='json'):
                     abort(404, message='No cached result found for this query.')
 
         if query_result:
+            if models.db.session.query(models.Query, models.DataSource).filter(
+                    models.Query.query_hash == query_result.query_hash,
+                    models.DataSource.type == 'result').first():
```

Comment:
I've been trying to move away from the session API, to be more explicit what it is we're querying against. Does this look for a query whose datasource type is "result"?

I think the SQL created is the same as with:

```python
models.Query.query.join(
    models.DataSource,
).filter(
    models.Query.query_hash == query_result.query_hash,
    models.DataSource.type == 'result',
).first()
```